### PR TITLE
chore(ftb-5) : Setup eslint,prettier,husky 

### PR DESCRIPTION
**Pull request by Duong-N**

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,4 +1,7 @@
 {
   "singleQuote": true,
-  "trailingComma": "all"
-}
\ No newline at end of file
+  "trailingComma": "all",
+  "printWidth": 100,
```
Comment (canhbk) [inline .prettierrc]: @Duong-N set printwidth to 80

File: .prettierrc [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
+  "tabWidth": 2,
```
Comment (canhbk) [inline .prettierrc]: @Duong-N use space with 4 size, instead of `tab` for the intent

**Review by canhbk — CHANGES_REQUESTED** [review]

**Comment by canhbk** [comment]
Update title of PR to correct format @Duong-N

File: .husky/pre-commit
Comment (canhbk) [inline .husky/pre-commit:1]: Run lint-staged in this pre-commit

File: .prettierrc [hunk 3]
```diff
@@ -1,4 +1,11 @@
 {
-  "singleQuote": true,
-  "trailingComma": "all"
-}
\ No newline at end of file
+    "trailingComma": "all",
+    "printWidth": 100,
+    "tabWidth": 4,
```
Comment (canhbk) [inline .prettierrc:4]: do we need this `tabWidth` anymore?

File: package.json [hunk 4]
```diff
@@ -1,88 +1,98 @@
...
+            "^.+\\.(t|j)s$": "ts-jest"
+        },
+        "collectCoverageFrom": [
+            "**/*.(t|j)s"
+        ],
+        "coverageDirectory": "../coverage",
+        "testEnvironment": "node"
+    },
+    "lint-staged": {
+        "*.ts": [
```
Comment (canhbk) [inline package.json:93]: should lint for other file extension as well: js, json, yml, etc

**Review by canhbk — CHANGES_REQUESTED** [review]

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        '@typescript-eslint/interface-name-prefix': 'off',
+        '@typescript-eslint/explicit-function-return-type': 'off',
+        '@typescript-eslint/explicit-module-boundary-types': 'off',
+        '@typescript-eslint/no-explicit-any': 'off',
+        'prettier/prettier': [
+            'error',
+            {
+                singleQuote: true,
+                trailingComma: 'all',
+                printWidth: 100,
```
Comment (canhbk) [inline .eslintrc.js]: set printWidth to 80

Comment (canhbk) [reply .eslintrc.js]: should not override prettier config in the eslint

**Review by canhbk — CHANGES_REQUESTED** [review]

**Comment by Duong-N** [comment]
image
Result

**Review by canhbk — APPROVED** [review]

**Pull request merged**